src/EditorFeatures/Test/Diagnostics/AbstractDiagnosticProviderBasedUserDiagnosticTest.cs

```diff
@@ -111,5 +114,14 @@ internal override IEnumerable<Diagnostic> GetDiagnostics(TestWorkspace workspace
                 }
             }
         }
+
+        private void AssertNoAnalyzerExceptionDiagnostics(IEnumerable<Diagnostic> diagnostics)
+        {
+            // The internal method `Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor::IsAnalyzerExceptionDiagnostic()`
+            // does essentially this, but due to linked files between projects, this project cannot have internals visible
```

Comment:
There is already a project-to-project reference to the `CodeAnalysis` project that is required, so adding the internals-visible-to stuff just exposes more types which would make the extern alias required all over the place making all of the code in this project look ugly.  I'm opting for this potentially overly agressive re-implementation over the original.